### ISSUE TITLE
polylang functions are not working in my unit test cases

### ISSUE THREAD

**Issue by benerd** [polylang/polylang#1069]
## Prerequisites

- [x] This is not a usage question (Those should be directed to the [community supported forum](https://wordpress.org/support/plugin/polylang), unless this is a question about Polylang Pro in which case you should [use the helpdesk](https://polylang.pro/support/)).
- [x] I have searched for similar issues in both open and closed tickets and cannot find a duplicate.
- [x] The issue still exists against the latest `master` branch of Polylang and the latest WordPress version.
- [x] This bug happens with only Polylang plugin active
- [ ] This bug happens with a default WordPress theme active
- [ ] I can reproduce this bug consistently

## Expected behavior and actual behavior

This is my bootstrap config to load the plugin. 
```function _manually_load_environment() {
	// Options.
	update_option( 'permalink_structure', '/blog/%postname%' );

	// Activate plugins.
	$plugins_to_activate = array(
		'polylang-pro/polylang.php'
	);
	update_option( 'active_plugins', $plugins_to_activate );

	// Switch theme.
	add_filter( 'stylesheet', function () {
		return 'theme-name';
	} );
}
tests_add_filter( 'muplugins_loaded', '_manually_load_environment' ); 
```

now when I activate the plugin using the above code tax query is appended in wp_query and language taxonomy is also added. when when I use any  pll function i.e. ```pll_current_language()``` it gives me the below error.

```Call to undefined function pll_current_language()```

so after some googling I came to know polylang deactiactivates itself if there is no language added. So I did the below configration.

```
public static function set_up_before_class() {
                $options = PLL_Install::get_default_options();
		$options['hide_default'] = 0; // Force option to pre 2.1.5 value otherwise phpunit tests break on Travis.
		$options['media_support'] = 1; // Force option to pre 3.1 value otherwise phpunit tests break on Travis.
		self::$model = new PLL_Admin_Model( $options );
		$languages = include POLYLANG_DIR . '/settings/languages.php';

		$locale    = 'en_US';
		$values    = $languages[ $locale ];

		$values['slug']       = 'us';
		$values['rtl']        = (int) ( 'rtl' === $values['dir'] );
		$values['term_group'] = 0;

		$args = [];
		$args = array_merge( $values, $args );

		$errors = self::$model->add_language( $args );
		if ( is_wp_error( $errors ) ) {
			throw new InvalidArgumentException( $errors->get_error_message() );
		} 
		self::$model->clean_languages_cache();```

}

public function set_up() {
		parent::set_up();
		self::$model->options['hide_default'] = 1;
		self::$model->options['post_types'] = array(
			'trcpt' => 'trcpt',
		);
		self::$model->options['taxonomies'] = array(
			'trtax' => 'trtax',
		);

		self::$model->post->register_taxonomy(); // needs this for 'lang' query var
		create_initial_taxonomies();
		register_post_type( 'trcpt', array( 'public' => true, 'has_archive' => true ) ); // translated custom post type with archives
		register_taxonomy( 'trtax', 'trcpt' ); // translated custom tax
		register_post_type( 'cpt', array( 'public' => true, 'has_archive' => true ) ); // *untranslated* custom post type with archives
		register_taxonomy( 'tax', 'cpt' ); // *untranslated* custom tax

		$links_model = self::$model->get_links_model();
		$links_model->init();

		$this->frontend = new PLL_Frontend( $links_model );
		$this->frontend->init();

}
```
but it still gives me the same error.

```Call to undefined function pll_current_language()```

**Comment by Chouby** [comment]
The file for PAI functions is loaded only if languages exist. See https://github.com/polylang/polylang/blob/3.2.5/include/class-polylang.php#L241.
So you must including in your tests after the languages are created. See for example: https://github.com/polylang/polylang/blob/3.2.5/tests/phpunit/tests/test-strings.php#L15

**Comment by benerd** [comment]
hey as you can see I am adding a language in the set_up_before_class () , I have also included the Api as well.
now it gives me this error =>  Attempt to read property "model" on null.

Is it because the language is not being added properly.

**Comment by Chouby** [comment]
Try adding `$GLOBALS['polylang'] = $this->frontend;` at the end of your `set_up()`. This global is the entry point to internal objects for most of the API functions.

**Comment by benerd** [comment]
Hey @Chouby thanks. I was able to access the functions after adding the above solution you have provided. But do I need to create the languages in all the test cases, I was checking the polylang unit test cases and you guys are extending the PLL_UnitTestCase in the test classes.

I am trying to do it in the bootstrap so that so that what I configure in the bootsrtap is available for all the testcases. so to achieve this, I am creating the setting up all this in my bootstrap. There I am able to add languages and if I print the languages using `pll_languages_list()` but if I am printing them in any of my test case I got no languages.
